qds_sdk/commands.py

```diff
@@ -1155,6 +1154,60 @@ def parse(cls, args):
         v["command_type"] = "DbTapQueryCommand"
         return v
 
+class ListCommands(Command):
+    usage = "listCmds [options]"
+    optparser = GentleOptionParser(usage=usage)
+    optparser.add_option("--status", dest="status",
+                         help="Status of the commands to list. [waiting, default=all]")
+    optparser.add_option("--page", dest="page",
+                         help="Page number.")
+    optparser.add_option("--per-page", dest="per_page",
+                         help="Results per page.")
+    @classmethod
+    def parse(cls, args):
+        """
+
+        Args:
+            `args`: sequence of arguments
+        Returns:
+            Dictionary that can be used in create method
+
+        Raises:
+            ParseError: when the arguments are not correct
+        """
+        try:
+            (options, args) = cls.optparser.parse_args(args)
+        except OptionParsingError as e:
+            raise ParseError(e.msg, cls.optparser.format_help())
+        except OptionParsingExit as e:
+            return None
+        v = vars(options)
+        v["command_type"] = "ListCommand"
+        return v
+
+    @classmethod
+    def list_action(cls, args):
+        """
+
+        For the given parameters(like status), return the list of commands.
+
+        Returns:
+            Json Unicode with command information.
+
+        """
+        conn = Qubole.agent()
+        params = {}
+        if args["status"] is not None:
+            params["status"] = args["status"]
+        else:
+            params["status"] = "done"
```

Comment:
Currently, I think the API is not working as expected and asked people to look into it. But for status waiting, it works fine. So I will hold on to the PR until it gets resolved. 
But, by default, I will do an all if it is possible.